tests/unit/amo/components/TestConfirmationDialog.js

```diff
@@ -42,77 +26,65 @@ describe(__filename, () => {
     expect(cancelButton).toHaveClassName('ConfirmationDialog-cancel-button');
     expect(cancelButton).toHaveProp('buttonType', 'cancel');
     expect(cancelButton.children()).toHaveText('Cancel');
+    */
   });
 
   it('renders a custom className', () => {
     const className = 'MyComponent';
-    const root = render({ className });
-    expect(root).toHaveClassName(className);
-    expect(root).toHaveClassName('ConfirmationDialog');
+    render({ className });
+    expect(screen.getByClassName(className)).toHaveClass('ConfirmationDialog');
   });
 
-  it.each([true, false])(
-    'can configure buttons as puffy=%s',
-    (puffyButtons) => {
-      const root = render({ puffyButtons });
-
-      expect(root.find('.ConfirmationDialog-cancel-button')).toHaveProp(
-        'puffy',
-        puffyButtons,
-      );
-      expect(root.find('.ConfirmationDialog-confirm-button')).toHaveProp(
-        'puffy',
-        puffyButtons,
-      );
-    },
-  );
+  it('can configure buttons as non puffy', () => {
+    render({ puffyButtons: false });
 
-  it('calls onConfirm() when user clicks Confirm', () => {
-    const onConfirm = sinon.spy();
-    const root = render({ onConfirm });
+    expect(
+      screen.getByClassName('ConfirmationDialog-cancel-button'),
```

Comment:
Should I still do `toHaveClass('ConfirmationDialog-cancel-button')` ? I feel it's important to ensure that too since we have styles depending on it too...

Reply:
Ah just read your other comment. I'm going to leave the assertion for now

Reply:
I did some more thinking about my comment and I think we *should not* be making assertions about hardcoded class names. There are a multitude of class names that have been added to elements in our code for which we don't have tests - I think by far the majority of them are not tested. In general, when existing tests reference class names it's mostly for locating an element, and sometimes it is to make an assertion about logic. If we are not already making assertions about the majority of hardcoded class names, I see no reason to start now.  What do you think?